src/main/java/com/nutomic/syncthingandroid/syncthing/RestApi.java

```diff
@@ -687,6 +702,111 @@ protected void onPostExecute(String s) {
+
+                        switch (eventType) {
+                            // This special shortcut can be used if data only contains strings.
+                            // It just copies everything into a bundle.
+                            case "itemfinished":
+                            case "foldercompletion":
+                            case "deviceconnected":
+                            case "devicediscovered":
+                            case "statechanged":
+                                dataBundle = new Bundle();
+                                JSONObject data = json.getJSONObject("data");
+
+                                for (Iterator<String> keyIterator = data.keys(); keyIterator.hasNext();) {
+                                    String key = keyIterator.next();
+                                    dataBundle.putString(key, data.getString(key));
+                                }
+
+                                // If the event contains a folder keyword but no path synthesise
+                                // a path keyword to ease processing of the event.
+                                String folder = data.optString("folder", null);
+                                if ((folder != null) && (data.optString("path", null) == null)) {
+                                    String folderPath = getPathForFolder(folder);
+                                    if (folderPath != null) {
+                                        dataBundle.putString("path", folderPath);
+                                    }
+                                }
+
+                                break;
+
+                            // Ignored events.
+                            case "ping":
+                                break;
+
+                            default:
+                                Log.d(TAG, "Unhandled event " + json.getString("type"));
+                        }
+
+                        if (dataBundle != null) listener.onEvent(id, eventType, dataBundle);
+                    }
+
+                    listener.onDone(lastId);
+                }
+                catch (JSONException e) {
+                    Log.w(TAG, "Failed to read events", e);
+                }
+            }
+        };
+
+        // Within honeycomb execute was fully parallel. Since honeycomb the AsyncTasks are serialized and if the event interface
+        // blocks (no new events), all other tasks will block also. To prevent this, starting with HONEYCOMB the thread pool executor must be used.
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB)
+            return new AsyncState(eventGetTask.execute(mUrl, GetTask.URI_EVENTS, mApiKey, "since", String.valueOf(sinceId), "limit", String.valueOf(limit)));
+        else
+            return new AsyncState(eventGetTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, mUrl, GetTask.URI_EVENTS, mApiKey, "since", String.valueOf(sinceId), "limit", String.valueOf(limit)));
```

Comment:
I must be missing something. minSdk is 11 (3.0, Honeycomb), meaning there's no need to cater for anything lower. The else clause here works on versions >= 11, not just > 11, which we indeed support.